### PR TITLE
[BottomNavigation] Support `accessibilityLabel`, `accessibilityHint`, and `isAccessibilityElement`

### PR DESCRIPTION
**Pull request by andrewplai**
BottomNavigation tab views were not reflecting some of the UIAccessibility properties of the UITabBarItem objects used to create them. This PR adds support for `accessibilityLabel`, `accessibilityHint`, and `isAccessibilityElement`, including Key-Value Observing (KVO) of changes to the UITabBarItem's properties.

### CODE REVIEW

**Review by romoore — CHANGES_REQUESTED** [review]
Thank you for the PR, andrewplai! Could you please add unit tests around the new behavior? I think we should also make sure that any changes to these values are observed in `addObserversToTabBarItems` and removed in `removeObserversFromTabBarItems`.

**Comment by andrewplai** [comment]
Sounds good.  Went ahead and added in some unit tests for the new behavior, and also made sure that the changes are observed in addObserversToTabBarItems / removeObserversFromTabBarItems.

File: components/BottomNavigation/tests/unit/BottomNavigationTests.m [hunk 1]
```diff
@@ -144,6 +144,46 @@ - (void)testAccessibilityIdentifier {
   XCTAssert([bar.itemViews.firstObject.accessibilityIdentifier isEqualToString:newIdentifier]);
 }
 
+-(void)testAccessibilityLabel {
+  NSString *oldLabel = @"oldLabel";
```
Comment (romoore) [inline components/BottomNavigation/tests/unit/BottomNavigationTests.m:162]: Please break these into tests for single behaviors and add  "Given", "When", and "Then" comments to the tests.  An example might be:

```objc
- (void)testAccessibilityHintInitialValue {
  // Given
  NSString *oldHint = @"oldHint";
  NSString *newHint = @"newHint";
  UITabBarItem *tabBarItem = [[UITabBarItem alloc] initWithTitle:@"Home"
                                                           image:nil
                                                             tag:0];
  tabBarItem.accessibilityHint = oldHint;
  MDCBottomNavigationBar *bar = [[MDCBottomNavigationBar alloc] init];
  // When
  bar.items = @[ tabBarItem ];

  // Then
  XCTAssert([bar.itemViews.firstObject.accessibilityHint isEqualToString:oldHint]);
}

(void)testAccessibilityHintValueChanged {
  // Given
  NSString *oldHint = @"oldHint";
  NSString *newHint = @"newHint";
  UITabBarItem *tabBarItem = [[UITabBarItem alloc] initWithTitle:@"Home"
                                                           image:nil
                                                             tag:0];
  tabBarItem.accessibilityHint = oldHint;
  MDCBottomNavigationBar *bar = [[MDCBottomNavigationBar alloc] init];
  bar.items = @[ tabBarItem ];

  // When
  tabBarItem.accessibilityHint = newHint;

  // Then
  XCTAssert([bar.itemViews.firstObject.accessibilityHint isEqualToString:newHint]);
}

**Review by romoore — APPROVED** [review]
Looks much better. A few nits about the tests, and I've kicked-off the continuous integration builds. I suspect clang-format will come back with some comments.  Once things are cleaned-up, this should be ready to merge. 👍

**Comment by romoore** [comment]
Looks good to me!

File: components/BottomNavigation/tests/unit/BottomNavigationTests.m (hunk 1 above)
Comment (andrewplai) [reply components/BottomNavigation/tests/unit/BottomNavigationTests.m:162]: Sounds good!  Went ahead and split the unit tests as suggested.

**Comment by andrewplai** [comment]
Hi, just wanted check on the status of when this pull request will be merged.  Let me know if I need to add in anything else.  Thanks!

**Comment by andrewoverton** [comment]
Thanks @andrewplai !

**Pull request merged**